app/scripts/migrations/142.1.ts

```diff
@@ -171,11 +171,6 @@ function transformState(
   const { allTokens } = tokensControllerState;
   const allTokensForChain = allTokens[currentChainId];
   if (!isObject(allTokensForChain)) {
```

Comment:
If we hit this case, what would `tokensControllerState.tokens` be?

Reply:
`tokensControllerState.tokens` should be empty array since there's no tokens on the current seleced chain

Reply:
Maybe we should still capture a sentry error if `isObject(allTokensForChain)` is false but `tokensControllerState.tokens` is not empty?

In case there is some other data corruption involved that we are not aware of?

Reply:
i can add that

Reply:
added here [93f7494](https://github.com/MetaMask/metamask-extension/pull/30163/commits/93f74947a4ad444b9e91d03b33561a3ac7ebec06)